test/common/Client.js

```diff
@@ -1707,6 +1758,12 @@ describe('The Client class', function() {
   });
 
   describe('The authPassword method', function() {
+    var provider = {
+      newPromise: function(resolver) {
+        return q.Promise(resolver);
+      }
+    };
```

Comment:
I'm pretty sure using `new jmap.QPromiseProvider()` below instead of this will work :)